src/test/test-capability.c

```diff
@@ -77,6 +86,34 @@ static int setup_tests(void) {
         test_uid = nobody->pw_uid;
         test_gid = nobody->pw_gid;
 
+        /* Find out the kernel version to determine if the ambient
+         * capability tests should be run. */
+
+        if (uname(&u) < 0)
+                return -errno;
+
+        version = strdup(u.release);
+        orig = version;
+
+        token = strsep(&version, ".");
+        if (!token || !version) {
+                return -EINVAL;
+        }
+
+        r = safe_atou(token, &major);
+        if (r < 0)
+                return r;
+
+        /* Can't use safe_atou here because the string is not clean. */
+        minor = strtoul(version, &end, 10);
+        if (version == end) {
+                return -EINVAL;
+        }
+
+        /* Only run ambient capability tests on kernel versions >= 4.3. */
+        if (major > 4 || (major == 4 && minor >= 3))
+                *run_ambient = true;
+
```

Comment:
Ah, wait. We should probably have a better way of detecting whether that feature is supported. After all, it could be that distros back-port certain features without bumping the kernel version string. I didn't look too deep yet, but isn't there a way to do a dummy call or something, in order to figure out the set of supported capability sets?